listen/property-changes.js

```diff
@@ -363,48 +449,30 @@ PropertyChanges.prototype.makePropertyObservable = function (key) {
 };
 
 // constructor functions
-
 PropertyChanges.getOwnPropertyChangeDescriptor = function (object, key) {
-    if (object.getOwnPropertyChangeDescriptor) {
-        return object.getOwnPropertyChangeDescriptor(key);
-    } else {
-        return PropertyChanges.prototype.getOwnPropertyChangeDescriptor.call(object, key);
-    }
+    return object.getOwnPropertyChangeDescriptor ? object.getOwnPropertyChangeDescriptor(key) : PropertyChangesP.getOwnPropertyChangeDescriptor.call(object, key);
 };
 
 PropertyChanges.hasOwnPropertyChangeDescriptor = function (object, key) {
-    if (object.hasOwnPropertyChangeDescriptor) {
-        return object.hasOwnPropertyChangeDescriptor(key);
-    } else {
-        return PropertyChanges.prototype.hasOwnPropertyChangeDescriptor.call(object, key);
-    }
+    return object.hasOwnPropertyChangeDescriptor ? object.hasOwnPropertyChangeDescriptor(key) : PropertyChangesP.hasOwnPropertyChangeDescriptor.call(object, key);
 };
 
 PropertyChanges.addOwnPropertyChangeListener = function (object, key, listener, beforeChange) {
-    if (!Object.isObject(object)) {
-    } else if (object.addOwnPropertyChangeListener) {
-        return object.addOwnPropertyChangeListener(key, listener, beforeChange);
-    } else {
-        return PropertyChanges.prototype.addOwnPropertyChangeListener.call(object, key, listener, beforeChange);
-    }
+	if(Object(object) === object) {
```

Comment:
This creates an object if the value is not an object already. Although clever and succinct, `object && typeof object === "object"` is a better pattern.